backend/vkt/src/main/java/fi/oph/vkt/service/PaymentService.java

```diff
@@ -0,0 +1,190 @@
+package fi.oph.vkt.service;
+
+import fi.oph.vkt.model.Enrollment;
+import fi.oph.vkt.model.ExamEvent;
+import fi.oph.vkt.model.Payment;
+import fi.oph.vkt.model.Person;
+import fi.oph.vkt.model.type.EnrollmentStatus;
+import fi.oph.vkt.model.type.PaymentStatus;
+import fi.oph.vkt.payment.paytrail.Customer;
+import fi.oph.vkt.payment.paytrail.Item;
+import fi.oph.vkt.payment.paytrail.PaytrailConfig;
+import fi.oph.vkt.payment.paytrail.PaytrailResponseDTO;
+import fi.oph.vkt.repository.EnrollmentRepository;
+import fi.oph.vkt.repository.PaymentRepository;
+import fi.oph.vkt.util.exception.APIException;
+import fi.oph.vkt.util.exception.APIExceptionType;
+import fi.oph.vkt.util.exception.NotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class PaymentService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(PaymentService.class);
+
+  private static final int COST_SINGLE = 45400;
+  private static final int COST_MULTIPLE = 22700;
+
+  private final PaytrailService paytrailService;
+  private final PaymentRepository paymentRepository;
+  private final EnrollmentRepository enrollmentRepository;
+  private final Environment environment;
+
+  private Item getItem() {
+    return Item
+      .builder()
+      .units(1)
+      .unitPrice(COST_MULTIPLE)
+      .vatPercentage(PaytrailConfig.VAT)
+      .productCode("foo") // TODO: Mikä productCode halutaan? Onko väliä?
+      .build();
+  }
+
+  private int getTotal(List<Item> itemList) {
+    // TODO: Selvitä halutaanko jokainen taito omana rivinä vai yhtenä
+    //    return itemList.size() > 1
+    //            ? COST_MULTIPLE
+    //            : COST_SINGLE;
+    return itemList.stream().reduce(0, (subtotal, item) -> subtotal + item.unitPrice(), Integer::sum);
+  }
+
+  private List<Item> getItems(final Enrollment enrollment) {
+    final List<Item> itemList = new ArrayList<>();
+
+    // TODO: Selvitä halutaanko jokainen taito omana rivinä vai yhtenä
```

Comment:
Oliko tähän tullut selvyyttä? YKIssä (tosin tarkistusarviontien puolella) on yksi rivi per osakoe. Rivi per taito kuulostaa mielestäni ihan hyvältä idealta, mutta voi olla että siitä aiheutuu myös ongelmia - esim. jos on valinnut sekä suullisen että kirjallisen taidon, myös ymmärtämisen taito taitaa tulla silloin kaupanpäällisenä -> ymmärtämisen taidolle tulisi tällöin vissiin "hinnaksi" 0e? Vai jätettäisiinkö tällaisessa tapauksessa ymmärtämisen taito kokonaan pois laskutettavista itemeistä?

Reply:
Kyseltiin tätä viime weeklyssä. Kuulosti, ettei asiakas oikein osaa tähän suoraa vastausta antaa. Kallistuttiin kuitenkin tuohon rivi per taito.

Voisi kai tehdä noin, että se tulee 0e arvoisena. En oo juuri tätä paytrailia aiemmin pyöritelly, niin en tiedä näkyykö nuo rivit käytännössä edes missään? Ainakaan jos paytrailillä ostaa jotain, niin paytrailin omassa sähköpostissa ei ole eritelty tuotteita.